### PR TITLE
[Playerglobewien]

### PR DESCRIPTION
**Pull request by Oneboy1979**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [X] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

- new extractor for globewien player
- download works as expected

### CODE REVIEW

**Comment by mawi1** [comment]
I think your extractor would also work for https://player.hader.at/. Could you add support for it?

**Comment by mawi1** [comment]
Cool. Thank you.

**Comment by dstftw** [comment]
Read coding conventions.

**Comment by Oneboy1979** [comment]
I'll have a look at this!

**Comment by Oneboy1979** [comment]
@dstftw Fixed in accordance with the coding convention.

**Comment by Oneboy1979** [comment]
@dstftw Fixed in accordance with the coding convention.

File: youtube_dl/extractor/playerglobewien.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+from .common import InfoExtractor
+
+
+class PlayerGlobeWienIE(InfoExtractor):
+    _VALID_URL = r'https?://player.(?:globe.wien|hader.at)/(?:globe-wien|hader)/(?P<id>.*)'
```
Comment (sehaas) [inline youtube_dl/extractor/playerglobewien.py:9]: maybe you could restrict group `id` to exclude query parameters etc.
```python
r'https?://player.(?:globe.wien|hader.at)/(?:globe-wien|hader)/(?P<id>[^/?#]+)
```
example: https://player.hader.at/hader/hader-homestory-teil-6?streamNow=true

**Comment by sehaas** [comment]
I also [started an extractor](https://github.com/sehaas/youtube-dl/tree/globe_player) (before checking open PRs :/ ) but took a slightly different approach. All required data is already included in the webpage and no additional API request is necessary.

File: youtube_dl/extractor/playerglobewien.py (hunk 1 above)
Comment (Oneboy1979) [reply youtube_dl/extractor/playerglobewien.py:9]: ok that is a good point, i'll implement it this way, thanks in advance

**Comment by Oneboy1979** [comment]
@sehaas i checked your code and think this is more flexible as mine, i also include to extract all thumbnails from given link, can i use your code in this PR?

**Comment by sehaas** [comment]
Yes, feel free to integrate my code.

File: youtube_dl/extractor/playerglobewien.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        next_data = self._parse_json(
+            self._search_regex(
+                r'<script[^>]+id="__NEXT_DATA__"[^>]+type="application/json"[^>]*>([^<]+)</script>',
+                webpage, 'next data'),
+            video_id)
+
+        vod = next_data.get('props').get('initialState').get('vod')
+
+        formats = []
+        for key in vod.get('streamUrl'):
```
Comment (sehaas) [inline youtube_dl/extractor/playerglobewien.py:68]: iterate over dictionary tuples
```python
for key, src_url in vod.get('streamUrl').items():
```

File: youtube_dl/extractor/playerglobewien.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                formats.extend(self._extract_mpd_formats(
+                    src_url, video_id, mpd_id=key, fatal=False))
+            else:
+                formats.append({
+                    'format_id': key,
+                    'url': src_url
+                })
+
+        thumbnails = []
+        for key in vod.get('images'):
```
Comment (sehaas) [inline youtube_dl/extractor/playerglobewien.py:83]: see other comment

**Comment by sehaas** [comment]
Hi @dstftw,
are there any objections left or can this extractor be merged?

**Comment by Oneboy1979** [comment]
Hi @dstftw,
is something wrong with this PR, or can you merge it?

regards Oneboy1979

**Comment by sehaas** [comment]
Hi @Oneboy1979, could you squash all these commits into a single clean commit. Maybe the PR gets merged then?

**Pull request closed without merging**